storage/index.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package storage
```

Comment:
Maybe `storage` is too generic? Could we rename this to `gcs` or something google-cloud specific?

Reply:
I was also thinking about `storage/v2`, as `v2` is strictly bound with GCP buckets. WDYT?